### ISSUE TITLE
`ishermitian` is too restrictive

### ISSUE THREAD

**Issue by GiggleLiu** [JuliaLang/julia#28885]
I was expecting the following code print two `true`s.
```
using LinearAlgebra
siz = 8
base_space = qr(randn(Float64, siz, siz)).Q
phases = rand(siz)
signs = Diagonal(phases)
A = base_space*signs*base_space'
println(A ≈ A')
println(ishermitian(A))
```
Instead, I get `false` and `true`.

The `ishermitian` turns out using `==`. Isn't it better to use `≈` or add an `atol` keyword parameter?
https://github.com/JuliaLang/julia/blob/4498d27c3c3e11effdfdfdce822d025079f01a0a/stdlib/LinearAlgebra/src/generic.jl#L971

**Comment by GiggleLiu** [comment]
I think this "feature" is not consistent with other julia methods such as `eigen`.
`eigen(A)` here will return **real** eigenvalues, which means the program take `A` as a hermitian matrix, although it has ~1e-16 error.

**Comment by KristofferC** [comment]
See the discussion in https://github.com/JuliaLang/julia/issues/10298.

**Comment by garrison** [comment]
I agree that the default behavior of `ishermitian` should not be changed (in line with previous discussion), but in my opinion an optional `atol` keyword argument would be a welcome feature.

**Comment by garrison** [comment]
> It's possibly less of a decision and more that no one has submitted a PR?

If this is the case, it could be clarified by having an open issue regarding this missing feature.  This could be accomplished most easily by opening and renaming this one, while perhaps adjusting the top comment (I no longer have the permission to do any of this), or alternatively by having a new issue elsewhere.

Suggested [edited] title: `ishermitian` should accept optional `atol` keyword parameter

**Comment by garrison** [comment]
Oh, interesting find regarding #36243.  I had been thinking recently that anything decided here should apply to `issymmetric` as well.

**Comment by oscardssmith** [comment]
`isapprox(A,A')` will probably be about 2x slower, but that's probably not a major problem for many use cases.

**Comment by jlapeyre** [comment]
I wrote a package a while back to try to approach specifying  measure of closeness in a uniform way: [IsApprox.jl](https://github.com/jlapeyre/IsApprox.jl). I included code for many functions/methods including `ishermitian`.
> Four subtypes of `AbstractApprox` are included: `Equal`, `Approx`, `EachApprox`, and `UpToPhase`.
IsApprox implements the interface at least partially for each of: isone, iszero, ishermitian, issymmetric, isreal, isinteger, istriu, istril, isbanded, isdiag, isposdef, ispossemidef, isunitary, isinvolution.
